device_calendar/example/lib/presentation/pages/calendar_event.dart

```diff
@@ -331,113 +332,148 @@ class _CalendarEventPageState extends State<CalendarEventPage> {
                       ),
                     ),
                     Padding(
-                      padding: const EdgeInsets.all(10.0),
-                      child: TextFormField(
-                        initialValue: _interval?.toString(),
-                        decoration: const InputDecoration(
-                            labelText: 'Interval between events',
-                            hintText: '1'),
-                        keyboardType: TextInputType.number,
-                        validator: _validateInterval,
-                        onSaved: (String value) {
-                          _interval = int.tryParse(value);
-                        },
+                      padding: const EdgeInsets.fromLTRB(15, 0, 15, 10),
+                      child: Row(
+                        children: <Widget>[
+                          Text('Repeat Every '),
+                          Flexible(
+                            child: TextFormField(
+                              initialValue: _interval?.toString() ?? '1',
+                              decoration: const InputDecoration(hintText: '1'),
+                              keyboardType: TextInputType.number,
+                              inputFormatters: [
+                                WhitelistingTextInputFormatter.digitsOnly, 
+                                LengthLimitingTextInputFormatter(2)
+                              ],
+                              validator: _validateInterval,
+                              textAlign: TextAlign.right,
+                              onSaved: (String value) {
+                                _interval = int.tryParse(value);
+                              },
+                            ),
+                          ),
+                          _recurrenceFrequencyToIntervalText(_recurrenceFrequency),
+                        ],
                       ),
                     ),
-                    if (_recurrenceFrequency == RecurrenceFrequency.Weekly ||
-                        _recurrenceFrequency == RecurrenceFrequency.Monthly ||
-                        _recurrenceFrequency == RecurrenceFrequency.Yearly)
-                      DaysOfTheWeekFormEntry(daysOfTheWeek: _daysOfTheWeek),
-                    if (_recurrenceFrequency ==
-                        RecurrenceFrequency.Monthly) ...[
+                    if (_recurrenceFrequency == RecurrenceFrequency.Weekly) ... [
+                      Column(
+                          children: [ 
+                            ...DayOfWeek.values.map((d) {
```

Comment:
Generally speaking I would try to avoid using one letter variables, as they are usually meaningless and make the code less readable. Not to say that we can't use them at all: https://softwareengineering.stackexchange.com/a/285293

Reply:
That was also already like that, I've updated all that I could find.
You can find the fix at this commit: ad0ca29